addons/dynamishelper/dynamishelper.lua

```diff
@@ -24,408 +24,427 @@
-   				initializeproc()
-   			end
-		end
-	end
-end)
-
-
+-------------------------------------------------------------------------------
+-- Parse the chat messages for drops, staggers and time extensions ------------
+-------------------------------------------------------------------------------
 windower.register_event('incoming text',function (original, new, color)
---	print('event_incoming_text function')
-	if timer == 'on' then
-  		a,b,fiend = string.find(original,"%w+'s attack staggers the (%w+)%!")
-   		if fiend == 'fiend' then
-			StaggerCount = StaggerCount + 1
-    		windower.send_command('timers c '..StaggerCount..' 30 down')
-    		return new, color
-    	end
-	end
- 	if tracker == 'on' then
-     	a,b,item = string.find(original,"%w+ obtains an? ..(%w+ %w+ %w+ %w+)..\46")
-     		if item == nil then
-      	 		a,b,item = string.find(original,"%w+ obtains an? ..(%w+ %w+ %w+)..\46")
-       				if item == nil then
-         				a,b,item = string.find(original,"%w+ obtains an? ..(%w+%-%w+ %w+)..\46")
-          					if item == nil then
-           						a,b,item = string.find(original,"%w+ obtains an? ..(%w+ %w+)..\46")
-         					end
-       				end
-     		end
--- 		a,b,item = string.find(original,"%w+ obtains an? ..(.*)..\46")
- 		if item ~= nil then
- 			item = item:lower()
- 			for i=1, #Currency do
-				if item == Currency[i]:lower() then
-   					Currency[Currency[i]] = Currency[Currency[i]] + 1
-   				end
- 			end
- 			obtainedf()
- 		end
-    	initializebox()
- 	end
- 	return new, color
-end)
-
-function obtainedf()
-	obtained = nil
- 	for i=1,#Currency do
- 		if Currency[Currency[i]] ~= 0 then
-  			if obtained == nil then
-  				obtained = " "
-  			end
-   			obtained = (obtained..Currency[i]..': '..Currency[Currency[i]]..' \n ')
- 		end
- 	end
-end
-
-windower.register_event('zone change', function(id)
-	goodzone = ProcZones:contains(id)
-	if not goodzone then
-		windower.text.set_visibility('proc_box', false)
-	end
+    local time = nil
+    local item = nil
+    local fiend = original:match("%w+'s attack staggers the (%w+)%!")
+    if settings.timer then
+        if fiend == 'fiend' then
+            windower.send_command('timers c "'..current_mob..'" 30 down stun')
+            return new, color
+        end
+    end
+    if string.find(original,"remaining in Dynamis.") then
+        obj_time = (tonumber(original:match("%d+")) * 60)
+        end_time = os.time() + obj_time
+    end
+    if string.find(original,"will be expelled from Dynamis") then
+        obj_time = (tonumber(original:match("%d+")) * 60)
+        end_time = os.time() + obj_time
+    end
+    if string.find(original,"Your stay in Dynamis has been extended by %d+ minutes.") then
+        end_time = end_time + (tonumber(original:match("%d+")) * 60)
+        w.time = os.date('!%H:%M:%S', end_time)
+    end
+    item = original:match("Obtained key item: ..(%w+ %w+ %w+ %w+)..\46")
+    if item ~= nil then
+        item = item:lower()
```

Comment:
This is also not your code, but there's really no reason to use `:lower()` on both strings. We get them directly from the game, so they should match case as well.